runtime/compiler/env/VMJ9.cpp

```diff
@@ -278,14 +278,14 @@ TR_J9VMBase::instanceOfOrCheckCastNoCacheUpdate(J9Class *instanceClass, J9Class*
    return ::instanceOfOrCheckCastNoCacheUpdate(instanceClass, castClass);
    }
 
-// Points to address: what if vmThread is not the compilation thread.
 // What if the compilation thread does not have the classUnloadMonitor.
 // What if jitConfig does not exist.
 // What if we already have the compilationShouldBeInterrupted flag set.
 // How do we set the error code when we throw an exception
 
-// IMPORTANT: acquireVMAccessIfNeeded could throw an exception,
-// hence it is important to call this within a try block.
+// IMPORTANT: acquireVMAccessIfNeeded could throw an exception if the vmThread
+// **IS** a Compilation Thread; hence it is important to call this within a try
+// block.
```

Comment:
TR_DisableNoVMAccess is an option that instruct the compilation threads to hold VM access during the compilation. It does not apply to other threads.
It seems that this method was written to specifically be called by compilation threads.

Reply:
I moved the code that checks the options to the point where we know it's a comp thread.